lib/kafka/producer_interceptors.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Kafka
+  # Holds a list of producer interceptors that implement on_send
+  # and wraps calls to a chain of custom producer interceptors.
+  class ProducerInterceptors
+    def initialize(interceptors:, logger:)
+      @interceptors = interceptors || []
+      @logger = TaggedLogger.new(logger)
+    end
+
+    # This method is called when the client produces a message, before it gets queued to be sent to the Kafka brokers.
+    # The message returned from the first ProducerInterceptor.on_send is passed to the second interceptor on_send, and so on in an
+    # interceptor chain. This method does not throw exceptions.
+    #
+    # @param pending_message [Kafka::PendingMessage] the produced message.
+    #
+    # @return [Kafka::PendingMessage] the intercepted message returned by the last interceptor.
+    def on_send(pending_message)
+      @interceptors.each do |interceptor|
+        begin
+          pending_message = interceptor.on_send(pending_message)
+        rescue Exception => e
+          @logger.warn "Error executing interceptor on_send for topic: #{pending_message.topic} partition: #{pending_message.partition} #{e.class}: #{e.message}\n#{e.backtrace.join("\n")}"
```

Comment:
The only difference between `ConsumerInterceptors` and this class is the error message here, right? I think we could use a single `Interceptors` class and either use a generic error message _or_ use some kind of parameter to `initialize` that we use to construct the error message.

Reply:
made the error message generic